optimizer/resolver.go

```diff
@@ -299,20 +308,28 @@ func (nr *nameResolver) resolveColumnNameInContext(ctx *resolverContext, cn *ast
 			}
 			return nr.resolveColumnInResultFields(cn, ctx.fieldList)
 		}
-		// Field list first, then from table.
+		// Resolve from table first, then from select list.
+		found := nr.resolveColumnInTableSources(cn, ctx.tables)
+		if nr.Err != nil {
```

Comment:
I am not sure about this. I think if we meet err, we should interrupt the visiting process. It is nothing to do with found.